### PR TITLE
Cherry-pick repo test fixes 

### PR DESCRIPTION
**Pull request by sambible**
cherrypicking some test fixes into 6.10.4

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repository.py -k test_positive_access_protected_repository

**Comment by akhil-jha** [comment]
LGTM

**Review by akhil-jha — APPROVED** [review]

**Comment by swadeley** [comment]
Hi @sambible ,  
when raising a PR for cherry-picks it is customary to provide links to the original PR in the initial comment.
thank you

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -1606,7 +1602,7 @@ def test_negative_synchronize_private_registry_no_passwd(
         with pytest.raises(
             HTTPError,
             match='422 Client Error: Unprocessable Entity for url: '
-            f'{default_sat.url}/katello/api/v2/repositories',
+            f'{default_sat.url}:443/katello/api/v2/repositories',
```
Comment (swadeley) [inline tests/foreman/api/test_repository.py:1605]: Hi @sambible , do you really need to add `443` here? That is is the default port for HTTPS and when you run the test you can see the scheme is "https":

    >>> default_sat.url                                                                    
    'https://<redacted>.redhat.com'

**Review by swadeley — APPROVED** [review]
ACK with one minor question.

File: tests/foreman/api/test_repository.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/api/test_repository.py:1605]: Ok, sorry, I see that is the error message you are checking, /o\

**Pull request merged**